### ISSUE TITLE
Support for auto-relax after `poetry add`

### ISSUE THREAD

**Issue by madkinsz** [madkinsz/poetry-relax#5]
It's unclear if Poetry would disapprove of this as their documentation states that core command behavior should not be changed but a maintainer notes  that a plugin may be a good way to change the default behavior in [this comment](https://github.com/python-poetry/poetry/issues/3427#issuecomment-1001973463).

It seems as though it would be straight-forward to add a hook after the `add` command runs that modifies the `pyproject.toml` using the mechanisms established for the `poetry relax` command. It's a little redundant to parse the file again, but I doubt the impact on speed will be noticeable.

If added, this would include a configurable toggle and would likely be off my default.

This issue will track requests for this feature. Feel free to add a 👍 or 👎 reaction here. Additional thoughts are welcome.

**Comment by madkinsz** [comment]
> This poetry-relax module at the moent could just be a single-line sed-script replace some chars in a file, after you ran "poetry add".

To provide some clarity here: Yes, you could use `sed` to replace some chars in a file. However, this plugin does far more than that. It leverages a lot of Poetry internals to provide a similar experience to using Poetry itself. It adds support for leveraging the Poetry dependency checker to guarantee that modifying the version specifications does not create an unsolvable dependency tree. It includes the ability to update the lock file. It allows the dependencies to be updated with a flag. These are not features that could be done well with `sed` and are the reason this library exists.

**Comment by zanieb** [comment]
If people are interested in trying this, I have a draft implementation at https://github.com/zanieb/poetry-relax/pull/49

I need to add test coverage, so expect there to be problems still.

**Comment by adam-grant-hendry** [comment]
@zanieb Yes, I think this would be a great addition. [`poetry-plugin-sort`](https://pypi.org/project/poetry-plugin-sort/) is one such plugin that sets a precedence for this. They utilize the exposed [`cleo` events](https://python-poetry.org/docs/plugins/#event-handler) to sort the dependencies after `poetry init` or `poetry add` are run.

If worried about the loose definition of "changing", `poetry-plugin-sort` also adds a `pyproject.toml` configuration variable that you can set to turn on or off sorting after `poetry init` or `poetry add` are implemented. I think this would be a wise addition since it would allow users to turn this feature on or off at their discretion.
